hc-dna/zomes/perspective_diff_sync/src/workspace.rs

```diff
@@ -408,20 +408,16 @@ impl Workspace {
                             branches.remove(branch_index);
                             //If there are no more branches and we have truly reached the end
                             search.reached_end = true;
-                            //NOTE: this if block is the code that breaks the test_latest_join tests, with it removed the tests pass, but test three null parents fails
-                            if branches.len() == 0 {
-                                if common_ancestor.is_none() && other.reached_end == true {
-                                    common_ancestor = Some(NULL_NODE());
-                                    self.terminate_with_null_node(current_hash, side, &mut searches)?;
-                                };
-                            }
+                            if common_ancestor.is_none() && other.reached_end == true && !check_if_vec_item_in_another_vec(branches, &other.found_ancestors.borrow()) {
+                                common_ancestor = Some(NULL_NODE());
+                                self.terminate_with_null_node(current_hash, side, &mut searches)?;
+                            };
```

Comment:
Just checking if get this right: this is about when we terminate the collecting with a null node, meaning we are sure there is no common ancestor.

We are at a place where our current diff doesn't have a parent, so we have reached a (local) end.

Before we said: if we don't have any other branches left AND if the other side also reached its end, we terminate.

Now: we don't check if we have no local branches left. Instead we check if any of our open branches (i.e. their first diff) is in other's found_ancestors. This means, we would terminate in a case where we do have open branches our our side, if other side reached end and our branch heads are not found on other's side..

correct?

if so: why?